internal/tableland/impl/mesa.go

```diff
@@ -35,150 +36,131 @@ func NewTablelandMesa(
 
 // ValidateCreateTable allows to validate a CREATE TABLE statement and also return the structure hash of it.
 // This RPC method is stateless.
-func (t *TablelandMesa) ValidateCreateTable(
-	ctx context.Context,
-	req tableland.ValidateCreateTableRequest,
-) (tableland.ValidateCreateTableResponse, error) {
+func (t *TablelandMesa) ValidateCreateTable(ctx context.Context, statement string) (string, error) {
 	ctxChainID := ctx.Value(middlewares.ContextKeyChainID)
 	chainID, ok := ctxChainID.(tableland.ChainID)
 	if !ok {
-		return tableland.ValidateCreateTableResponse{}, errors.New("no chain id found in context")
+		return "", errors.New("no chain id found in context")
 	}
-	createStmt, err := t.parser.ValidateCreateTable(req.CreateStatement, chainID)
+	createStmt, err := t.parser.ValidateCreateTable(statement, chainID)
 	if err != nil {
-		return tableland.ValidateCreateTableResponse{}, fmt.Errorf("parsing create table statement: %s", err)
+		return "", fmt.Errorf("parsing create table statement: %s", err)
 	}
-	return tableland.ValidateCreateTableResponse{StructureHash: createStmt.GetStructureHash()}, nil
+	return createStmt.GetStructureHash(), nil
 }
 
 // ValidateWriteQuery allows the user to validate a write query.
-func (t *TablelandMesa) ValidateWriteQuery(
-	ctx context.Context,
-	req tableland.ValidateWriteQueryRequest,
-) (tableland.ValidateWriteQueryResponse, error) {
+func (t *TablelandMesa) ValidateWriteQuery(ctx context.Context, statement string) (tables.TableID, error) {
 	ctxController := ctx.Value(middlewares.ContextKeyAddress)
 	controller, ok := ctxController.(string)
 	if !ok || controller == "" {
-		return tableland.ValidateWriteQueryResponse{}, errors.New("no controller address found in context")
+		return tables.TableID{}, errors.New("no controller address found in context")
 	}
 
 	ctxChainID := ctx.Value(middlewares.ContextKeyChainID)
 	chainID, ok := ctxChainID.(tableland.ChainID)
```

Comment:
I was considering making the controller address and chain ID explicit parameters to this func and one other that does the same thing. This would feel more correct to me for a "core API", no hidden surprises. I left it as is to get some input from reviewers. What do you think?

Reply:
Agree, I think it is a good idea.

I think the `ctx` for storing values is usually a slippery slope and too opaque. Moreover, I think the `ctx` use here comes from the fact that this service was tied to RPC, and usually these values come from middleware that can only send them injected in `ctx`.

Technically speaking, if we think about the CoreAPI as untangled from RPCs, is up to the caller to use whatever mechanism they can, and unpack that complexity there while passing "plain arguments" at this layer.

I'm fine too with creating a `TblContext` struct if we think that's worth it to avoid parameter bloat. I don't think that's the case yet since we have about two usual parameters (`ChainID` and `Caller`). No strong opinions.

Reply:
Ok those params are now explicit. Was able to clean up a bit of cruft in the tests because of this.